packages/swapper/src/swappers/RelaySwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,242 @@
+    affiliateBps,
+    potentialAffiliateBps,
+    slippageTolerancePercentageDecimal: _slippageTolerancePercentageDecimal,
+  } = input
+
+  const slippageToleranceBps = _slippageTolerancePercentageDecimal
+    ? convertDecimalPercentageToBasisPoints(_slippageTolerancePercentageDecimal).toFixed()
+    : undefined
+
+  const sellRelayChainId = relayChainMap[sellAsset.chainId]
+  const buyRelayChainId = relayChainMap[buyAsset.chainId]
+
+  if (sellRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (buyRelayChainId === undefined) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  if (!sendAddress) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'sendAddress is required',
+      }),
+    )
+  }
+
+  const maybeQuote = await getQuote(
+    {
+      originChainId: sellRelayChainId,
+      originCurrency: getRelayEvmAssetAddress(sellAsset),
+      destinationChainId: buyRelayChainId,
+      destinationCurrency: getRelayEvmAssetAddress(buyAsset),
+      tradeType: 'EXACT_INPUT',
+      amount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      recipient: receiveAddress,
+      user: sendAddress,
+      slippageTolerance: slippageToleranceBps,
+    },
+    deps.config,
+  )
+
+  if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
+
+  const quote = maybeQuote.unwrap()
+
+  const swapSteps = quote.data.steps.filter(step => step.id !== 'approve')
+  const hasApprovalStep = quote.data.steps.find(step => step.id === 'approve')
+
+  if (swapSteps.length > 2) {
+    deps.mixPanel?.track(MixPanelEvent.RelayMultiSteps, {
+      swapper: SwapperName.Relay,
+      method: 'get',
+    })
+
+    return Err(
+      makeSwapErrorRight({
+        message: `Relay quote with ${swapSteps.length} swap steps not supported (maximum 2)`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  const slippageTolerancePercentageDecimal = (() => {
+    if (_slippageTolerancePercentageDecimal) return _slippageTolerancePercentageDecimal
+    const destinationSlippageTolerancePercentageDecimal = bnOrZero(
+      quote.data.details?.slippageTolerance?.destination?.percent,
+    )
+
+    if (destinationSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(destinationSlippageTolerancePercentageDecimal).toFixed()
+    }
+
+    const originSlippageTolerancePercentageDecimal = bnOrZero(
+      quote.data.details?.slippageTolerance?.origin?.percent,
+    )
+
+    if (originSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(originSlippageTolerancePercentageDecimal).toFixed()
+    }
+  })()
```

Comment:
q: since we only support single hops, is the assumption that there will ever only be slippage on *one* of the two of origin/destination?

Reply:
Yep this is the case: 
- Same chain => using the origin
- multi chain => using the destination